bmap-rs/src/main.rs

```diff
@@ -89,12 +92,30 @@ fn setup_input(path: &Path) -> Result<Decoder> {
     }
 }
 
-fn copy(c: Copy) -> Result<()> {
-    if !c.image.exists() {
-        bail!("Image file doesn't exist")
+async fn setup_remote_input(url: Url, path: PathBuf) -> Result<Response> {
+    if path.extension().unwrap() != "gz" {
+        bail!("Image file format not implemented")
     }
+    reqwest::get(url).await.map_err(anyhow::Error::new)
```

Comment:
does the await wait for the whole file to download to memory ?

Reply:
We talked about it here: https://github.com/collabora/bmap-rs/pull/41#discussion_r1006696117
I was following this suggestion:
```
    let r = reqwest::get("http://collabora.com").await.unwrap();
    let mut b = r
        .bytes_stream()
        .map_err(|e| std::io::Error::new(std::io::ErrorKind::Other, e))
        .into_async_read();
```